### PR TITLE
Remove Mocks + Separate Adapters mapping

### PR DESCRIPTION
**Pull request by JTraversa**
Initially, adapters was a unique mapping per market (address underlying -> uint256 maturity -> uint8 protocol -> address adapter.

From this I figured it would be best to fold them into the market, (address underlying -> uint256 maturity -> market.adapters).

But in the process we (@alexanderwende) found some weirdness in the implementation, and that if some just sit in the marketplace per protocol, perhaps they all should.

So this PR removes `adapters` from the `Market` struct, and creates a `uint8 -> address adapters` mapping within the Marketplace contract.

Also in line with some previous convos the mocks just happened to be removed in this PR

### CODE REVIEW

File: src/Lender.sol [hunk 1]
```diff
@@ -549,7 +549,7 @@ contract Lender {
         IMarketPlace.Market memory _Market = IMarketPlace(marketplace).markets(u, m);
 
         // Conduct the lend operation to acquire principal tokens
-        (bool success, bytes memory returndata) = _Market.adapters[p].delegatecall(
+        (bool success, bytes memory returndata) = IMarketPlace(marketplace).adapters(p).delegatecall(
```
Comment (alexanderwende) [inline src/Lender.sol:552]: Here you are using `IMarketPlace(marketplace).adapters` (the interface)

File: src/Lender.sol [hunk 2]
```diff
@@ -599,7 +599,7 @@ contract Lender {
         // If the lst parameter is not populated, a swap is not required
         if (lst == address(0)) {
             // Conduct the lend operation to acquire principal tokens
-            (success, returndata) = _Market.adapters[p].delegatecall(
+            (success, returndata) = MarketPlace(marketplace).adapters(p).delegatecall(
```
Comment (alexanderwende) [inline src/Lender.sol:602]: ... and here you are using `MarketPlace(marketplace).adapters` (the contract)

Any specific reason for choosing 2 different versions of acquiring the adapters?

File: src/Lender.sol [hunk 3]
```diff
@@ -627,7 +627,7 @@ contract Lender {
                 require (msg.value >= spent, 'Insufficient ETH');
             }
             // Conduct the lend operation to acquire principal tokens
-            (success, returndata) = _Market.adapters[p].delegatecall(
+            (success, returndata) = MarketPlace(marketplace).adapters(p).delegatecall(
```
Comment (alexanderwende) [inline src/Lender.sol:630]: ... and here it's `MarketPlace(marketplace).adapters` too...

File: src/MarketPlace.sol [hunk 4]
```diff
@@ -35,9 +35,17 @@ contract MarketPlace {
         Notional // 8
     }
 
+    mapping(uint8 => address) public adapters;
+
+    // method sets adapters using an array of uint8's and array of addresses
+    function setAdapters(address[] calldata a) external {
```
Comment (alexanderwende) [inline src/MarketPlace.sol:41]: Should this method be `authorized`?

Comment (JTraversa) [reply src/MarketPlace.sol:41]: Oops yup, I had already fixed this locally but hadnt pushed up!

File: src/Lender.sol (hunk 2 above)
Comment (JTraversa) [reply src/Lender.sol:602]: It looks like it was included initially by sourabh for the `Principals` Enum, which I think I can include in the marketplace interface itself to work around that.

**Pull request merged**